### PR TITLE
Rewrite commentlock.py

### PR DESCRIPTION
**Pull request by yuwex**
It seems like this code was based on scratchclient, which isn't written in the cleanest, safest, or most efficient way. Since I plan to use this script, I rewrote it to be a bit more commented and to follow more best-practices.

* Removed unnecessary `json`, `re`, and `math` modules for faster execution.
* Broke scripts into segmented parts—one for setting up a session and one for changing comment access.
* Used more efficient code (Removing regex (or re) saves a bunch of time, also added f-strings wherever possible)
* Made code run via `__name__ == "__main__"` which allows the code to be used by other modules via `from commentlock import prepare_session` or `from commentlock import set_comments_allowed` 
* Added `getpass` module to protect password from wandering eyes :)
* Changed print statements to be more descriptive
* Added optional cooldown timer to prevent automatic scratch bans when changing the status on large amounts of projects
* Added code-style black

### CODE REVIEW

File: commentlock.py [hunk 1]
```diff
@@ -1,92 +1,124 @@
...
+
+    # Make comments_on variable into 1-line
+    comments_on = input("Should comments be enabled? Y/n: ").lower() == "y"
+
+    # Add cooldown
+    cooldown = float(
+        input("How long should the request cooldown be? (0.1 recommended): ")
+    )
+
+    print("\n")
```
Comment (DogCatPuppyLover) [inline commentlock.py:113]: Is there a reason that this is on a different line?

File: commentlock.py [hunk 2]
```diff
@@ -1,92 +1,124 @@
...
+        offset += 20
+
+
+# Use __name__ == '__main__' to prevent accidental running
+if __name__ == "__main__":
+
+    print("Welcome to the commentlock tool!")
+
+    username = input("Enter your username: ")
+    password = getpass("Enter your password: ")
```
Comment (DogCatPuppyLover) [inline commentlock.py:103]: Is there a way to copy-and-paste your password into the getpass field? When I tried copy-and-pasting my password, it threw an error, but when I typed it out, it worked fine.

File: commentlock.py [hunk 3]
```diff
@@ -1,92 +1,124 @@
...
+
+    # Make comments_on variable into 1-line
+    comments_on = input("Should comments be enabled? Y/n: ").lower() == "y"
+
+    # Add cooldown
+    cooldown = float(
```
Comment (DogCatPuppyLover) [inline commentlock.py:109]: By automatic bans, do you mean account bans, or temporary request limits?

File: commentlock.py [hunk 4]
```diff
@@ -1,92 +1,124 @@
...
-    # Get CSRF token
-    r = s.get("https://scratch.mit.edu/csrf_token/", headers=headers)
-    print(r.status_code)
-    csrf_token = re.search(
-            "scratchcsrftoken=(.*?);", r.headers["Set-Cookie"]
-    ).group(1)
+    # Log in with CSRF Token.
+    resp = session.post(
+        "https://scratch.mit.edu/login/",
+        json=payload,
```
Comment (DogCatPuppyLover) [inline commentlock.py:38]: Should the equals sign have spaces around it?

**Review by DogCatPuppyLover — COMMENTED** [review]
This looks great! Thanks for contributing. I asked someone else to review it as well before I commit.

**Comment by DogCatPuppyLover** [comment]
Also, what is your Scratch username, so I can credit you?

File: commentlock.py (hunk 1 above)
Comment (yuwex) [reply commentlock.py:113]: Nope! If you want to combine the lines, feel free. I used the code stylizer, `black` to do all of the stylizing after I wrote the code, so it wasn't intentional.

File: commentlock.py (hunk 2 above)
Comment (yuwex) [reply commentlock.py:103]: I don't think so. Getpass emulates the password prompt that comes up whenever you try to run things as root via the command line, and you can't paste passwords there either.

File: commentlock.py (hunk 3 above)
Comment (yuwex) [reply commentlock.py:109]: I mean possible bans.

Scratch's ratelimiting is weird, and it is possible to get IP banned if you send too many requests (ex: repl.it servers can't use the /login/ endpoint anymore)

The cooldown is more of a precaution. If someone's request library could somehow send 100 requests per second (which is **highly** unlikely), the Scratch team would have grounds to ban that person due to the potential server damage spamming requests can cause (ex: the chat filter incident that happened a while ago).

File: commentlock.py (hunk 4 above)
Comment (yuwex) [reply commentlock.py:38]: The spaces are purely stylistic. I ran this code through [black](https://black.readthedocs.io/en/stable/), and it decided to remove the spaces as well as all of the other formatting.

**Comment by yuwex** [comment]
Could you credit `SansStudios` instead of yuwex? I realize that that account somehow still has new scratcher status :P

**Pull request merged**